index.ts

```diff
@@ -18,6 +18,13 @@ declare namespace ethereum {
 /** Host IPFS interface */
 export declare namespace ipfs {
   function cat(hash: string): Bytes | null
+  function map(hash: string, callback: string, flags: string[]): void
```

Comment:
I realized a problem with the API (and, consequently, `mapJSON` as well): As soon as people start using this, they will want to pass parameters from the calling side to the callback.

I think we may have to add another parameter either between the `callback`/`flags` or at the end: `userData: Value`. I'm thinking `Value` here because that's our low-level type that can represent pretty much anything. What do you think?

Reply:
Yes, that would make sense. Is there anything to worry about since the `Value` will be passed from one `WasmiModule` to another?

Reply:
Nope, it's basically being copied across.

Reply:
It would be easy and performant to pass those values through the store, the annoyance being that you have to add a type to the schema.

Reply:
Why do they need to be passed through the store? Isn't it just that when you "spawn" a n `ipfs.map` stream, that you copy the user data value and then pass it to every invocation of the handler?

Reply:
Yes, I'm just discussing the alternative workaround.

Reply:
Ah